### DISCUSSION TITLE
Support Linux hardware monitor kernel API

### DISCUSSION THREAD

**Discussion by HowardZorn** [NVIDIA/open-gpu-kernel-modules#580] [Ideas]
[AMDGPU](https://docs.kernel.org/gpu/amdgpu/thermal.html#hwmon-interfaces) have supported hwmon interfaces, which is very convenient for monitoring GPU utilization. Could NVIDIA add support for these interfaces as well?

**Comment by Athanasius** [comment]
I second this.

Rather than any developer needing to implement an extra API (i.e. the nvidia-smi/ NVML stuff), if the nvidia drivers implemented the `hwmon` interface then anything that already uses such would automatically gain access to the monitoring (and possibly setting of things like fan speeds?) without further work.

**Comment by Steinliiippp** [comment]
I second this too.

Would be much easier for current and future applications to interconnect with NVIDIA GPUs in a standardized way.

**Comment by mtijanic** [comment]
At the risk of stating the obvious, one of the reasons why this wasn't considered before is because the proprietary driver does not get access to symbols needed for the hwmon API. Nowadays that the default is the open driver that's less of a concern, but we do still try to maintain parity between them where possible (and also FreeBSD, Solaris, etc). So while it would be a nice feature to have, I can't really speak to which priority it might be looked at.

I imagine we wouldn't reject it if it came in a PR though :)

---

By the way, I once - just for the heck of it - hacked this up using FUSE+NVML. Unfortunately FUSE doesn't allow for overlay-like mounting, so the only way I got it working is by mounting on an existing `/sys/class/hwmon/hwmonX` node which meant losing one of the existing ones. Fun exercise, but not something that is actually widely usable.

**Comment by edpil02** [comment]
There a  PR for hwmon .

https://github.com/NVIDIA/open-gpu-kernel-modules/pull/1169
